src/channels/media.rs

```diff
@@ -1373,3 +1534,115 @@ where
+        channels::tests::MockTcpStream,
+        DEFAULT_RECEIVER_ID, DEFAULT_SENDER_ID,
+    };
+    use protobuf::EnumOrUnknown;
+
+    use super::*;
+
+    #[test]
+    fn test_get_status() {
+        let mut stream = MockTcpStream::new();
+        let payload = format!(
+            r#"{{
+            "requestId":1,
+            "type":"{}",
+            "status":[
+                {{
+                    "mediaSessionId":1,
+                    "playerState":"PLAYING",
+                    "playbackRate":1.0,
+                    "supportedMediaCommands":2300
+                }}
+            ]
+        }}"#,
+            MESSAGE_TYPE_MEDIA_STATUS
+        );
+        stream.set_message(crate::cast::cast_channel::CastMessage {
+            protocol_version: Some(EnumOrUnknown::new(ProtocolVersion::CASTV2_1_2)),
+            source_id: Some(DEFAULT_RECEIVER_ID.to_string()),
+            destination_id: Some(DEFAULT_SENDER_ID.to_string()),
+            namespace: Some(CHANNEL_NAMESPACE.to_string()),
+            payload_type: Some(EnumOrUnknown::new(PayloadType::STRING)),
+            payload_utf8: Some(payload),
+            payload_binary: None,
+            continued: None,
+            remaining_length: None,
+            special_fields: Default::default(),
+        });
+        let channel = MediaChannel {
+            sender: Cow::from(DEFAULT_SENDER_ID),
+            message_manager: Lrc::new(MessageManager::new(stream)),
+        };
+
+        let result = channel.get_status("MyAppTransportId", None).unwrap();
+
+        assert_eq!(1, result.request_id);
+        if let Some(entry) = result.entries.first() {
+            assert_eq!(1, entry.media_session_id);
+            assert_eq!(PlayerState::Playing, entry.player_state);
+            assert_eq!(1.0, entry.playback_rate);
+            assert_eq!(2300, entry.supported_media_commands);
+        }
+    }
+
+    #[test]
+    fn test_parse_media_error() {
+        let message = CastMessage {
+            namespace: CHANNEL_NAMESPACE.to_string(),
+            source: DEFAULT_RECEIVER_ID.to_string(),
+            destination: DEFAULT_SENDER_ID.to_string(),
+            payload: CastMessagePayload::String(
+                "{\"type\":\"ERROR\",\"detailedErrorCode\":104,\"itemId\":1}".to_string(),
+            ),
+        };
+        let channel = MediaChannel {
+            sender: Cow::from(DEFAULT_SENDER_ID),
+            message_manager: Lrc::new(MessageManager::new(MockTcpStream::new())),
+        };
+        let expected_result = MediaError {
+            detailed_error_code: MediaDetailedErrorCode::MediaSrcNotSupported,
+            message_type: MESSAGE_TYPE_ERROR.to_string(),
+        };
+
+        let response = channel.parse(&message).unwrap();
+
+        if let MediaResponse::Error(result) = response {
+            assert_eq!(expected_result, result);
+        } else {
+            panic!("expected MediaResponse::Error, but got {:?}", response);
+        }
```

Comment:
__note:__ it'd probably make sense to implement `PartialEq` for `MediaResponse` and related structs to make assertions easier, e.g.:
```rust
assert_eq!(
    Ok(MediaResponse::Error(expected_result)),
    channel.parse(&message)
);
```

But it's outside the scope of this PR, it's just the `panic` that caught my eye.